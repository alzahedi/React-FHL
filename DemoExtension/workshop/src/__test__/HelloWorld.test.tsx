import * as React from "react";
import { render, screen } from "@testing-library/react";
import HelloWorld from "../HelloWorld.ReactView";
import { act } from "react-dom/test-utils";

test("renders HelloWorld Component contains text Content", () => {
  render(<HelloWorld />);
  expect(screen.getByText("Hello World!")).toBeInTheDocument();
});


test("renders HelloWorld Component snapshot test", async () => {

  await act( async () => expect(render(<HelloWorld />)).toMatchSnapshot());
 ;
});