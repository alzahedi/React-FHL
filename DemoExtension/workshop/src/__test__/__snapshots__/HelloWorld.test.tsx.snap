// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders HelloWorld Component snapshot test 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    
        
    <div
      class="reactview-printonly"
      id="__frameTitle"
      style="height: 64px;"
    >
      
            
      <div
        style="display: flex; height: 100%; box-sizing: border-box; padding: 8px 20px;"
      >
        
                
        <span
          style="display: flex; align-items: center;"
        >
          
                    
          <img
            aria-hidden="true"
            id="__bladeIcon"
            style="height: 28px; width: 28px; margin-right: 10px;"
          />
          
                
        </span>
        
                
        <span
          style="flex-grow: 1; font-family: Segoe UI;"
        >
          
                    
          <div
            style="display: flex;"
          >
            
                        
            <h2
              id="__bladeTitle"
              style="margin:0; font-size: 24px; line-height: 28px; height: 32px; flex-grow: 1;"
            >
              
                            
              <span
                id="__bladeTitleMain"
                style="font-weight: 600;"
              />
              
                            
              <span
                id="__bladeTitleExtras"
                style="font-weight: 400;"
              />
              
                        
            </h2>
            
                    
          </div>
          
                    
          <div
            id="__bladeSubtitle"
            style="font-size: 12px; line-height: 14px; height: 16px; color: #646464;"
          />
          
                
        </span>
        
            
      </div>
      
        
    </div>
    
        
    <div
      id="__bladeShield"
      style="visibility: hidden"
    >
      
            
      <div
        style="position: fixed; top: 0; left: 0; bottom: 0; right: 0; z-index: 1000; background-color: inherit;"
      >
        
                
        <div
          class="fxs-progress-dots"
        >
          
                    
          <div
            class="fxs-progress-dots-dot"
          />
          
                    
          <div
            class="fxs-progress-dots-dot"
          />
          
                    
          <div
            class="fxs-progress-dots-dot"
          />
          
                    
          <div
            class="fxs-progress-dots-dot-shadow"
          />
          
                    
          <div
            class="fxs-progress-dots-dot-shadow"
          />
          
                    
          <div
            class="fxs-progress-dots-dot-shadow"
          />
          
                
        </div>
        
            
      </div>
      
        
    </div>
    
        
    <div
      id="root-container"
    >
      <div
        id="__frameTop"
      >
        
    
        <div
          id="__bladeCommandBar"
        />
        
    
        <div
          id="__bladeStatusBar"
        />
        
    
        <div
          id="__bladeDialog"
        />
        

      </div>
      

      <div
        id="root"
      />
      

      <div
        id="__bladeFooter"
      />
    </div>
    
        
    <div
      id="react-perftools"
      style="display: none; height: 415px;"
    />
    
    
    <div>
      <span
        class="css-110"
      >
        Hello World!asdadsdda
      </span>
      <div>
        <div>
           Hello 
          Unknown
        </div>
        <div>
          Your email ID is 
          Unknown
        </div>
      </div>
      <div>
        <i
          aria-hidden="true"
          class="root-111"
          data-icon-name="FavoriteStarFill"
        />
        <i
          aria-hidden="true"
          class="root-111"
          data-icon-name="FavoriteStarFill"
        />
        <i
          aria-hidden="true"
          class="root-111"
          data-icon-name="FavoriteStarFill"
        />
        <i
          aria-hidden="true"
          class="root-111"
          data-icon-name="FavoriteStarFill"
        />
        <i
          aria-hidden="true"
          class="root-111"
          data-icon-name="FavoriteStar"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <span
      class="css-110"
    >
      Hello World!asdadsdda
    </span>
    <div>
      <div>
         Hello 
        Unknown
      </div>
      <div>
        Your email ID is 
        Unknown
      </div>
    </div>
    <div>
      <i
        aria-hidden="true"
        class="root-111"
        data-icon-name="FavoriteStarFill"
      />
      <i
        aria-hidden="true"
        class="root-111"
        data-icon-name="FavoriteStarFill"
      />
      <i
        aria-hidden="true"
        class="root-111"
        data-icon-name="FavoriteStarFill"
      />
      <i
        aria-hidden="true"
        class="root-111"
        data-icon-name="FavoriteStarFill"
      />
      <i
        aria-hidden="true"
        class="root-111"
        data-icon-name="FavoriteStar"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
